core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcher.scala

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.mesos
+
+import org.apache.spark.{Logging, SecurityManager, SparkConf}
+import org.apache.spark.util.{SignalLogger, IntParam, Utils}
+
+import java.io.File
+import java.util.concurrent.{TimeUnit, CountDownLatch}
+
+import org.apache.spark.deploy.mesos.ui.MesosClusterUI
+import org.apache.spark.deploy.rest.MesosRestServer
+import org.apache.spark.scheduler.cluster.mesos._
+import org.apache.spark.deploy.mesos.MesosClusterDispatcher.ClusterDispatcherArguments
+import org.apache.mesos.state.{ZooKeeperState, InMemoryState}
+
+/*
+ * A dispatcher actor that is responsible for managing drivers, that is intended to
+ * used for Mesos cluster mode.
+ * This class is needed since Mesos doesn't manage frameworks, so the dispatcher acts as
+ * a daemon to launch drivers as Mesos frameworks upon request.
+ */
+private [spark] class MesosClusterDispatcher(
+    args: ClusterDispatcherArguments,
+    conf: SparkConf) extends Logging {
+
+  def dispatcherPublicAddress(conf: SparkConf, host: String): String = {
```

Comment:
minor: this can just be called `publicAddress`. It's pretty clear that this is the dispatcher.